File: python/151-200/186. Reverse Words in a String II.py
class Solution:
    # @param s, a list of 1 length strings, e.g., s = ['h','e','l','l','o']
    # @return nothing
    def reverseWords(self, s):
        s.reverse()
        start = 0
        for i in range(len(s)):
            if s[i] ==  " ":
                s[start:i] = reversed(s[start:i])
                start = i + 1
        s[start:] = reversed(s[start:])

File: cpp/151-200/186. Reverse Words in a String II.cc
class Solution {
public:
    void reverseWords(string &s) {
    	if(s.empty())
    		return;
    	reverse(s.begin(), s.end());
    	int i = 0;
    	while(i < s.size()){
    		int j = i;
    		while(j < s.size() && s[j] != ' '){
    			j++;
    		}
    		reverse(s.begin() + i, s.begin() + j);
    		i = j + 1;
    	}
    }
};